javalib/src/main/scala/java/net/URI.scala

```diff
@@ -135,13 +135,15 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
     case _ => false
   }
 
-  def getAuthority(): String = _authority.map(decodeComponent).orNull
+  def getAuthority(): String =
+    _authority.filter(_ != "").map(decodeComponent).orNull
```

Comment:
@gzm0, is there some particular combination that you saw that I could be missing?